domain-management/src/main/java/org/jboss/as/domain/management/access/AccessIdentityResourceDefinition.java

```diff
@@ -178,35 +176,40 @@ protected boolean requiresRuntime(OperationContext context) {
         }
     }
 
-    private static FutureTask<SecurityDomain> toFutureTask(InjectedValue<SecurityDomain> injectedValue) {
-        return new FutureTask<>((Callable<SecurityDomain>)  (() -> injectedValue.getValue()));
-    }
-
-    private static SecurityDomain toSecurityDomain(final FutureTask<SecurityDomain> futureTask) {
-        try {
-            return futureTask.get();
-        } catch (InterruptedException | ExecutionException e) {
-            throw new IllegalStateException(e);
-        }
-    }
-
     static class IdentityService implements Service<Void> {
 
-        private List<FutureTask<SecurityDomain>> getSecurityDomainFutures = Collections.emptyList();
+        private List<InjectedValue<SecurityDomain>> inflowSecurityDomains = Collections.emptyList();
+        private InjectedValue<SecurityDomain> configuredSecurityDomain = null;
+        private final ManagementSecurityIdentitySupplier securityIdentitySupplier;
+
+        public IdentityService(ManagementSecurityIdentitySupplier securityIdentitySupplier) {
+            this.securityIdentitySupplier = securityIdentitySupplier;
+        }
 
         @Override
         public void start(StartContext context) throws StartException {
-            if (getSecurityDomainFutures != null) {
-                getSecurityDomainFutures.forEach(FutureTask::run);
+             securityIdentitySupplier.setConfiguredSecurityDomainSupplier(configuredSecurityDomain::getValue);
+            if (inflowSecurityDomains != null && !inflowSecurityDomains.isEmpty()) {
+                try {
+                    securityIdentitySupplier.setInflowSecurityDomainSuppliers(
+                            inflowSecurityDomains.stream().map(f -> (Supplier<SecurityDomain>) (() -> f.getValue())).collect(Collectors.toList()));
```

Comment:
IDEA complains that

() -> f.getValue()

can be replaced with

f::getValue

No big deal; it's just a style thing. I wouldn't mention it except if you're touching the file anyway for the line below...  It's nice to have those kind of style warning gone in my IDE. :)